dev/bots/docs.sh

```diff
@@ -1,17 +1,55 @@
 #!/bin/bash
 set -e
 
+function deploy {
+  local total_tries="$1"
+  local remaining_tries=$(($total_tries - 1))
+  shift
+  while [[ "$remaining_tries" > 0 ]]; do
+    (cd "$FLUTTER_ROOT/dev/docs" && firebase deploy --token "$FIREBASE_TOKEN" --project "$@") && break
+    remaining_tries=$(($remaining_tries - 1))
+    echo "Error: Unable to deploy documentation to Firebase. Retrying in five seconds... ($remaining_tries tries left)"
+    sleep 5
+  done
+
+  [[ "$remaining_tries" == 0 ]] && {
+    echo "Command still failed after $total_tries tries: '$@'"
+    return 1
+  }
+  return 0
+}
+
+function script_location() {
+  local script_location="${BASH_SOURCE[0]}"
+  # Resolve symlinks
+  while [[ -h "$script_location" ]]; do
+    DIR="$(cd -P "$( dirname "$script_location")" >/dev/null && pwd)"
+    script_location="$(readlink "$script_location")"
+    [[ "$script_location" != /* ]] && script_location="$DIR/$script_location"
+  done
+  echo "$(cd -P "$(dirname "$script_location")" >/dev/null && pwd)"
+}
+
+# So that users can run this script from anywhere and it will work as expected.
+SCRIPT_LOCATION="$(script_location)"
+FLUTTER_ROOT="$(dirname "$(dirname "$SCRIPT_LOCATION")")"
```

Comment:
Add "assuming that the relative path between FLUTTER_ROOT and this script remains constant"?